### PR TITLE
fix: hide quick action if the application has old manifest structure

### PR DESCRIPTION
**Pull request by nahmed22**
hide the manifest change actions from the quick action list, if the manifest of the underlying app has array structure.

### CODE REVIEW

File: packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts [hunk 1]
```diff
@@ -25,7 +25,8 @@ export class ToggleSemanticDateRangeFilterBar
 
     async initialize(): Promise<void> {
         const isUI5VersionNotSupported = await areManifestChangesSupported();
-        if (isUI5VersionNotSupported) {
+        const pagesStructureInManifest = this.context.manifest['sap.ui.generic.app'].pages;
+        if (isUI5VersionNotSupported || Array.isArray(pagesStructureInManifest)) {
```
Comment (voicis) [inline packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts:29]: This check should be part of `areManifestChangesSupported` function, so we have one function that we can use to check if manifest changes can be used.

**Review by voicis — CHANGES_REQUESTED** [review]

File: packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts (hunk 1 above)
Comment (nahmed22) [reply packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts]: done

Comment (voicis) [reply packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts]: We shouldn't use `not` in function names. In this particular case we need the inverse value, but it is quite likely that we would also need to check if the manifest changes are supported and we would need to use double negation, which is just looks weird and is more confusing `!areManifestChangesNotSupported()`. `!areManifestChangesSupported()` is a lot easier to read and understand.

**Review by voicis — APPROVED** [review]
- Code changes looks good
- Review comments addressed
- Good test coverage
- Did not test locally

**Review by vadson71 — APPROVED** [review]
Changes look good
Coverage is good

**Pull request merged**